cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -201,10 +202,28 @@ private void stopWatchWait(JsonArray externalAuthenticationURIs) throws Exceptio
       port = configuration.getInt(Constants.Security.AUTH_SERVER_BIND_PORT);
     }
 
+    String announceURLs = configuration.get(Constants.Security.AUTH_SERVER_ANNOUNCE_URLS);
+    // If the announceURLs is set in configuration, add the URL's to the list
+    if (announceURLs != null) {
+      String[] urls = announceURLs.split(",");
+      for (String url : urls) {
+        String urlWithToken = String.format("%s/%s", url, GrantAccessToken.Paths.GET_TOKEN);
+        externalAuthenticationURIs.add(new JsonPrimitive(urlWithToken));
+      }
+      return;
+    } else {
+      String announceAddress = configuration.get(Constants.Security.AUTH_SERVER_ANNOUNCE_ADDRESS_DEPRECATED);
+      if (announceAddress != null) {
+        String url = String.format("%s://%s:%d/%s", protocol, announceAddress, port,
```

Comment:
Better check if the `announceAddress` contains pattern `.*:[0-9]+` to decide whether need to add the `port` or not.